### PR TITLE
Fix typo, and note BoundObjectRef isn't always checked

### PR DESCRIPTION
**Pull request by cceckman**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

- "Object" was missing a "c".
- Only the API server can be asserted to have access to the bound object; an arbitrary relying party doesn't necessarily have access to the object in the API server that is named in the claim. Note this in the `BoundObjectRef` field's documentation.

**Which issue(s) this PR fixes**:
None

**Special notes for your reviewer**:
Thanks!

**Does this PR introduce a user-facing change?**:

```release-note
NONE

```

Per discussion:

/cc @mikedanese 

Thanks!

### CODE REVIEW

**Comment by cceckman** [comment]
/remove-kind api-change
/kind documentation

This just changes the comments - it doesn't change any existing contracts, just documents them better.

**Comment by mourya007** [comment]
can we say this is the kind cleanup now ?

File: pkg/apis/authentication/types.go [hunk 1]
```diff
@@ -135,7 +135,9 @@ type TokenRequestSpec struct {
 	ExpirationSeconds int64
 
 	// BoundObjectRef is a reference to an object that the token will be bound to.
-	// The token will only be valid for as long as the bound objet exists.
+	// The token will only be valid for as long as the bound object exists.
+	// NOTE: The API server will validate the BoundObjectRef, but other audiences
+	// may not. Keep ExpirationSeconds small if you want prompt revocation.
```
Comment (mikedanese) [inline pkg/apis/authentication/types.go]: If you want this to make it into api docs, you need to make the same changes in:

staging/src/k8s.io/api/authentication/v1
staging/src/k8s.io/api/authentication/v1beta1

Then run make update.

File: pkg/apis/authentication/types.go [hunk 2]
```diff
@@ -135,7 +135,9 @@ type TokenRequestSpec struct {
 	ExpirationSeconds int64
 
 	// BoundObjectRef is a reference to an object that the token will be bound to.
-	// The token will only be valid for as long as the bound objet exists.
+	// The token will only be valid for as long as the bound object exists.
+	// NOTE: The API server will validate the BoundObjectRef, but other audiences
```
Comment (mikedanese) [inline pkg/apis/authentication/types.go]: s/API server/TokenReview/ just to be more explicit?

File: pkg/apis/authentication/types.go (hunk 1 above)
Comment (cceckman) [reply pkg/apis/authentication/types.go]: Done.

**Comment by cceckman** [comment]
Addressed comments; @mikedanese PTAL & TY!

**Pull request merged**